### PR TITLE
fix: Allow user overwritte default widget opts

### PR DESCRIPTION
**Pull request by Czaki**
This PR fixes https://github.com/napari/napari/issues/6354

It also renamems multiple variables starting from `_` to ending with `_` as it is convention to use leading underscore as an unused variable when it is a local variable in functions so make it harder for me to follow flow of code.

### CODE REVIEW

**Comment by Czaki** [comment]
It looks like a failure of the docs is unrelated to PR.

**Comment by tlambert03** [comment]
Since the pr is mostly rename now, can you point out which part specifically fixes the napari issue ?

File: src/magicgui/type_map/_type_map.py [hunk 1]
```diff
@@ -269,14 +269,14 @@ def _pick_widget_type(
...
         return wdg, options
 
-    _widget_type = match_type(_type, value)
-    if _widget_type:
-        _cls, opts = _widget_type
-        return _cls, {**options, **opts}
+    widget_type_ = match_type(type_, value)
+    if widget_type_:
+        cls_, opts = widget_type_
+        return cls_, {**opts, **options}
```
Comment (Czaki) [inline src/magicgui/type_map/_type_map.py:275]: this

File: src/magicgui/type_map/_type_map.py [hunk 2]
```diff
@@ -252,15 +252,15 @@ def _pick_widget_type(
...
 
     if is_result:
-        _widget_type = match_return_type(_type)
-        if _widget_type:
-            _cls, opts = _widget_type
-            return _cls, {**options, **opts}
+        widget_type_ = match_return_type(type_)
+        if widget_type_:
+            cls_, opts = widget_type_
+            return cls_, {**opts, **options}
```
Comment (Czaki) [inline src/magicgui/type_map/_type_map.py:263]: and here same change, but it is not the code branch reached by the original issue.

**Comment by Czaki** [comment]
the change is to change `{**options, **opts}` to `{**opts, **options}`

**Comment by tlambert03** [comment]
thanks!

**Pull request merged**